vendor/github.com/containers/libhvee/pkg/kvp/kvp.go

```diff
@@ -0,0 +1,144 @@
+//go:build linux
+
+package kvp
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+// readKvpData reads all key-value pairs from the hyperv kernel device and creates
+// a map representation of them
+func readKvpData() (KeyValuePair, error) {
+	ret := make(KeyValuePair)
+	for i := 0; i < 5; i++ {
+		// We need to seed the poolids
+		ret[PoolID(i)] = ValuePairs{}
+	}
+	kvp, err := unix.Open(KernelDevice, unix.O_RDWR|unix.O_CLOEXEC|unix.O_NONBLOCK, 0)
+	if err != nil {
+		return nil, err
+	}
+	defer unix.Close(kvp)
+
+	var (
+		hvMsg    hvKvpMsg
+		hvMsgRet hvKvpMsgRet
+	)
+
+	const sizeOf = int(unsafe.Sizeof(hvMsg))
+
+	var (
+		asByteSlice    = (*(*[sizeOf]byte)(unsafe.Pointer(&hvMsg)))[:]
+		retAsByteSlice = (*(*[sizeOf]byte)(unsafe.Pointer(&hvMsgRet)))[:]
+	)
+
+	hvMsg.kvpHdr.operation = OpRegister1
+
+	l, err := unix.Write(kvp, asByteSlice)
+	if err != nil {
+		return nil, err
+	}
+	if l != sizeOf {
+		return nil, ErrUnableToWriteToKVP
+	}
+
+next:
+	for {
+		var pfd unix.PollFd
+		pfd.Fd = int32(kvp)
+		pfd.Events = unix.POLLIN
+		pfd.Revents = 0
+
+		howMany, err := unix.Poll([]unix.PollFd{pfd}, Timeout)
+		if err != nil {
+			if err == unix.EINTR {
+				continue
+			}
+			return nil, err
+		}
+
+		if howMany == 0 {
+			return ret, nil
+		}
+
+		l, err := unix.Read(kvp, asByteSlice)
+		if err != nil {
+			if err == unix.EAGAIN || err == unix.EINTR || err == unix.EWOULDBLOCK {
```

Comment:
Can we get a comment on the why we are ignoring these errors?

Reply:
Okay, submitted in https://github.com/containers/libhvee/pull/38.  Let's not block this PR on it, though?

Reply:
Sure, that sounds good.